src/query/block/meta.go

```diff
@@ -43,3 +48,82 @@ func (m Metadata) Equals(other Metadata) bool {
 func (m Metadata) String() string {
 	return fmt.Sprintf("Bounds: %v, Tags: %v", m.Bounds, m.Tags)
 }
+
+// ResultMetadata describes metadata common to each type of query results,
+// indicating any additional information about the result.
+type ResultMetadata struct {
+	// LocalOnly indicates that this query was executed only on the local store.
+	LocalOnly bool
+	// Exhaustive indicates whether the underlying data set presents a full
+	// collection of retrieved data.
+	Exhaustive bool
+	// Warnings is a list of warnings that indicate potetitally partial or
+	// incomplete results.
+	Warnings []Warning
+}
+
+// NewResultMetadata creates a new result metadata.
+// NB: capacity for warning is initialized to 1, since it is unlikely that more
+// than one warning will exist.
+func NewResultMetadata() ResultMetadata {
+	return ResultMetadata{
+		LocalOnly:  true,
+		Exhaustive: true,
+		Warnings:   make([]Warning, 0, 1),
+	}
+}
+
+// CombineMetadata combines two result metadatas.
+func (m ResultMetadata) CombineMetadata(other ResultMetadata) ResultMetadata {
+	combinedWarnings := make([]Warning, 0, len(m.Warnings)+len(other.Warnings))
```

Comment:
Maybe only allocate if `len(m.Warnings)+len(other.Warnings)` is greater than zero?